core/src/main/scala/akka/persistence/cassandra/journal/TagWriter.scala

```diff
@@ -200,17 +200,18 @@ import scala.concurrent.duration._
           }
       }
       log.debug(s"Tag write complete. {}", summary)
-      if (awaitingFlush.isDefined) {
-        log.debug("External flush request")
-        if (sortedBuffer.nonEmpty) {
-          // FIXME, break into batches
-          write(sortedBuffer, Vector.empty[(Serialized, TagPidSequenceNr)], tagPidSequenceNrs, awaitingFlush)
-        } else {
-          sender() ! FlushComplete
-          context.become(idle(sortedBuffer, tagPidSequenceNrs))
-        }
-      } else {
-        flushIfRequired(sortedBuffer, tagPidSequenceNrs)
+      awaitingFlush match {
+        case Some(replyTo) =>
+          log.debug("External flush request")
+          if (sortedBuffer.nonEmpty) {
+            // FIXME, break into batches
+            write(sortedBuffer, Vector.empty[(Serialized, TagPidSequenceNr)], tagPidSequenceNrs, awaitingFlush)
+          } else {
+            replyTo ! FlushComplete
```

Comment:
here is the bug, `replyTo` vs `sender()`

Reply:
:+1: